generators/fe_defines.c

```diff
@@ -93,9 +93,27 @@ static void to_upper(char *s) {
 }
 
 static void sanitize_name(char *s) {
+  char *valid_before = "$valid$";
+  char *valid_after = "valid";
+  char *valid_found = NULL;
+
+  while ((valid_found = strstr(s, valid_before)) != NULL) {
```

Comment:
why don't you just use `memmove` to remove the `$` signs instead of making a copy?